### PR TITLE
Redesign BlazePress Campaigns for mobiles (2nd attempt)

### PR DESCRIPTION
**Pull request by timur987**
Related to https://jira.tumblr.net/browse/SELFSERVE-506
Related to the PR https://github.com/Automattic/wp-calypso/pull/77909 (It was abandoned because of conflicts with trunk)

⚠️ We decided to temporary remove showing the number of Impressions and Clicks because this data comes from Druid database and it is not stable at the moment. To make it stable we need to switch getting that data to DSP's DB. So, when the DSP backed is ready, we will return showing these numbers.

## Proposed Changes

* Address stats data structure provided by the campaigns endpoint in Campaign type
* Make Campaign Tab table columns responsive
* Add mobile version for Campaigns Tab
* We decided to hide stats related columns (Impressions and Clicks for now because we have no backend optimized to showing those columns)

## Testing Instructions

* Build assets (`yarn && yarn start`)
* If you do not have campaigns in production, you can use your local DSP
* Try different screen sizes from 320px and beyond

How it might work (the video shows Impressions and Clicks columns, but the current version does not include them, and it's intentional):

https://github.com/Automattic/wp-calypso/assets/115007291/e9e1fd54-8ff8-4ea5-b02b-77a282b46a2c

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/my-sites/promote-post-i2/components/campaigns-filter/index.tsx [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import { useTranslate } from 'i18n-calypso';
 import SegmentedControl from 'calypso/components/segmented-control';
 
-export type CampaignsFilterType = 'all' | 'active' | 'created' | 'rejected';
+export type CampaignsFilterType = 'all' | 'active' | 'created' | 'finished' | 'rejected';
```
Comment (sbarbosa) [inline client/my-sites/promote-post-i2/components/campaigns-filter/index.tsx:4]: Thanks for adding this new State!

**Review by sbarbosa — APPROVED** [review]
The code looks good 👍

I noticed changes to the `landing/stepper` package in this PR. Did we make those changes? It's probably safer to undo those changes before we merge this.

I found two details while testing:
1- The filter bar is bigger than expected, I think we are missing a max-length there.
Screenshot 2023-06-12 at 22 00 53


2- In the empty campaign page, in the design, the status filter is hidden (I would go a step further and try to hide the search bar too). We can treat this as an improvement because the posts page probably behaves similarly. (we can discuss/change this in a future PR.
Screenshot 2023-06-12 at 22 05 26

**Comment by timur987** [comment]
Hey @sbarbosa! Thanks for the review.

> I noticed changes to the landing/stepper package in this PR. Did we make those changes? It's probably safer to undo those changes before we merge this.

I also saw those changes but now (after merging with trunk once again) they are gone (since they are already in trunk).

> 1- The filter bar is bigger than expected, I think we are missing a max-length there.

Oh, sure, I forgot to adjust grid settings (since I've added another filter item).

> 2- In the empty campaign page, in the design, the status filter is hidden (I would go a step further and try to hide the search bar too). We can treat this as an improvement because the posts page probably behaves similarly. (we can discuss/change this in a future PR.

Are you sure? According to [Figma](https://www.figma.com/file/uvh2KtY65b7hGBHI9exbJ1/BlazePress?type=design&node-id=6424-57158&t=l3f88Y1C94WanHjU-0), both search and filter are shown, at least here:
image

The last thing I want to mention is that the campaigns filter has old style but I will address it in a separate PR.
image

**Comment by sbarbosa** [comment]
> Are you sure? According to [Figma](uvh2KtY65b7hGBHI9exbJ1-fi-6424_57158), both search and filter are shown, at least here:

I think I got confused, in some of the mobile pages the filter is gone, but I think that is a mistake in the figma. It should behave similarly to desktop. It's good the way you developed.

Tested it again and it's looking great! 👍

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/8043117

Hi @timur987, could you please edit the description of this PR and add a screenshot for our translators? Ideally it'd include all of the following strings:
- `%s click`
- `Open details`
- `Campaign`
- `Ends`
- `Completed`

Thank you in advance!

**Comment by timur987** [comment]
Hey @a8ci18n!

Sure!

## Strings from a mobile view

- `%s clicks` - the number of clicks that were produced by the advertising campaign;
- `Open details` - a link to view the Campaign's details.

Screenshot 2023-06-13 at 20 55 36

## String from desktop view:

- `Campaign` - the name of the column for storing the name of the advertising campaign;
- `Ends` - the name of the column for storing the campaign's end date;
- `Completed` - on of the campaigns' statuses that emphasize that campaigns successfully finished.

Screenshot 2023-06-13 at 21 01 18

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/8043117).

**Pull request merged**